### DISCUSSION TITLE
FeynCalc wishlist

### DISCUSSION THREAD

**Discussion by vsht** [FeynCalc/feyncalc#61] [Ideas]
This is a thread to collect some input from the users regarding the still missing functionality. 
Of course, I have my own thoughts on this, but perhaps I'm missing something that is badly needed since years.
Especially since I'm not working on many HEP topics (e.g. BSM) where FeynCalc is especially popular

Some features that are on top of my personal to-do list are

- Proper topology identification for multi-loop topologies that should work both for graphs and explicit propagators.
- UF-Parametrization for tensor and scalar integrals: [WIP](https://github.com/FeynCalc/feyncalc/pull/58)
- QGRAF interface
- A simple interface for offloading selected operations to FORM

**Comment by rolfmertig** [comment]
For educational and possibly also collaborational use I would like to port FeynCalc to the Wolfram Enterprise Private Cloud, i.e., a pure web-interface. If I find time and maybe interest I could probably do that in the near future.

**Comment by huansouza** [comment]
Is it possible to extend the functionalities for dimensions other then four? I think that it would be great if we could use feyncalc to do some computations in three dimensions, for example.

**Reply by vsht** [reply]
Actually 3 and D-1 dimensions are already supported: https://arxiv.org/pdf/2006.15451.pdf
Or am I perhaps missing your point?

**Reply by huansouza** [reply]
Oh, I wasn't aware of that. Thanks a lot.

**Reply by vsht** [reply]
The only problem is that it is not so well tested. Apart from myself and my colleagues from Munich,
I don't think that so far anyone has tried to calculate something in 3D using FeynCalc 9.3. 
So bug reports and suggestions are very welcome.

**Reply by huansouza** [reply]
I see. Vladyslav, I saw that in https://arxiv.org/pdf/2006.15451.pdf you make available some notebooks, do you know where I can find more notebooks which use FeynCalc to compute something in 3D? I would like to learn how to use it.

**Comment by vsht** [comment]
https://arxiv.org/src/2006.15451v1/anc/UsageExamples.nb
https://github.com/FeynCalc/feynonium/tree/master/Examples

But it depends on what you mean with 3D. Probably not nonrelativistic EFT calculations
that I primarily had in mind...

**Reply by huansouza** [reply]
I'm actually interested in models with a Chern-Simons term, just like: https://arxiv.org/pdf/hep-th/0305043.pdf

I'm also interested in higher dimensional models, see: 10.1103/PhysRevD.102.045004

I use FeynRules+FeynArts+FeynCalc to compute the feynman diagrams. In the 3D dimensional case, for exemple, I had some problems with the definition of the Pauli matrices, which are defined in 4D. Besides that, I had some problems when I tried to substitute D->3-2\epsilon or D->6-2\epsilon.

**Reply by vsht** [reply]
This is indeed quite different from nonrelativistic. NR means that you treat temporal and spatial indices separately,
but the spacetime dimension still remains 4 or D. What you refer to is a relativistic and covariant  calculation
in a space with only 2 spatial dimensions. 

In principle, as long as everything is covariant, many evaluations should work the same way. The main exceptions
being the epsilon tensor and g^5, where you will need to define something on your own. 
I don't think that you need to introduce Pauli matrices here as your reference defines Dirac matrices that 
obey the usual anticommutation relation ${g^mu,g^nu} = 2 eta^{mu nu}$.

But yes, in general, such a calculation would require a lot of care and cross-checks.

**Reply by vsht** [reply]
Since this a proper forum, I created a new thread for discussing this type of calculations:
https://github.com/FeynCalc/feyncalc/discussions/63
Otherwise the wishlist thread kind of gets difficult to navigate.

**Reply by huansouza** [reply]
Ok. Thanks a lot for the clarifications!

**Comment by SiWang4phys** [comment]
Feyncal has a lot of SM examples.
But are there examples of applying the model files generated by Feynrules?
and how it applies in FeynArts and FeynCalc.
It would be more convenient if we could use any models we created.
